lms/static/scripts/frontend_apps/components/SubmitGradeForm.tsx

```diff
@@ -101,17 +130,21 @@ export default function SubmitGradeForm({
 
   // Track if current grade has changed compared to what was originally loaded
   const hasUnsavedChanges = useMemo(
-    () => draftGradeValue !== null && draftGradeValue !== grade.data,
-    [draftGradeValue, grade.data]
+    () =>
+      (draftGradeValue !== null && draftGradeValue !== grade.data?.grade) ||
+      (draftCommentValue !== null && draftCommentValue !== grade.data?.comment),
+    [draftCommentValue, draftGradeValue, grade.data?.comment, grade.data?.grade]
   );
 
   // Make sure instructors are notified if there's a risk to lose unsaved data
   useWarnOnPageUnload(hasUnsavedChanges);
 
-  // Clear the previous grade when the user changes.
+  // Clear the previous grade and hide comment controls when the user changes.
   useEffect(() => {
     setGradeSaved(false);
+    setShowCommentControls(false);
     setDraftGradeValue(null);
+    setDraftCommentValue(null);
   }, [student]);
```

Comment:
Just reusing this `useEffect` for now. It's already discussed that this should be handled differently, but let's keep that out of this PR for simplicity.